### DISCUSSION TITLE
Translating Edlib can now be done using Weblate

### DISCUSSION THREAD

**Discussion by tmkarlsen** [cerpus/Edlib#633] [Announcements]
Edlib is now ready for you to contribute by translating all user-facing parts of the application. Head over to https://weblate.edlib.com/projects/ and get started! Everyone can take part! 
Documentation for translation will be updated shortly: https://docs.edlib.com/docs/developers/translation

**Comment by samumist** [comment]
Thank you, Edlib team, for this wonderful project! 
However, the Weblate URL provided for the translation doesn't seem to be working. Is there another way I can contribute a Simplified Chinese translation?

**Reply by tmkarlsen** [reply]
Hi @samumist!

We are delighted to hear that you like Edlib! 
Since the major refactoring of Edlib was merged to master, [setting up Weblate again](https://github.com/cerpus/Edlib/issues/2899) has been a pending task.
In the meantime, we are happy to accept pull requests. The translation files are found here:

- https://github.com/cerpus/Edlib/tree/master/sourcecode/hub/lang
- https://github.com/cerpus/Edlib/tree/master/sourcecode/apis/contentauthor/resources/lang
- https://github.com/cerpus/Edlib/tree/master/sourcecode/apis/contentauthor/resources/assets/react/language

**Comment by samumist** [comment]
Hi @tmkarlsen !

Thank you for the prompt reply and guidance! I’ll prepare the pull requests with the translation.

**Quick question on language variants**

In the H5P content type, there are `zh`, `zh-cn`, and `zh-hans` variants for Simplified Chinese. Could you clarify which one Edlib uses by default, or should I provide all three for consistency? Your advice would help ensure my translations align with the project.

Appreciate your support!

Best regards,

Sam

**Reply by tmkarlsen** [reply]
We recommend using `zh-hans` as this is the correct one for Simplified Chinese.

We are excited to receive your contribution!

**Reply by samumist** [reply]
Hi @tmkarlsen,
Thanks!

I've just submitted the [pull request](https://github.com/cerpus/Edlib/pull/3073) with the `zh-hans` language files. 

I've been testing the translations locally with a Docker setup (screenshot attached).
<img width="1350" alt="截屏2025-03-31 20 20 59" src="https://github.com/user-attachments/assets/c783d7fe-e973-4ceb-bb8f-6526ef0bb1f2" />

I noticed a few CA interface strings are missing from the language files (screenshot attached).
<img width="1350" alt="截屏2025-03-31 20 21 59" src="https://github.com/user-attachments/assets/9515a855-b8e4-4f45-bc88-52f6210dc88e" />
<img width="1350" alt="截屏2025-03-31 20 21 35" src="https://github.com/user-attachments/assets/cd12accf-255f-4972-b341-32e24105015c" />

The translations for "Context" might not be perfect yet, as I'm still learning how "Context" works in Edlib. I'll refine those once I understand it better.

**Reply by tmkarlsen** [reply]
Hi, @samumist!

Nice work!  We'll take a look at it!

As you have discovered, the admin interface for Content Author (the LTI tool for H5P authoring) has not yet been made available for translations. This is not a prioritized part for translations, and we cannot tell you when it will be available.

We will look into the other missing parts of general admin interface and see if we can make those translatable.

**Reply by samumist** [reply]
Hi @tmkarlsen,

I hope you're doing well!
I'm currently working on translating the English text to Chinese in H5P Hub, but I’m unsure how to proceed. A screenshot of the relevant section is attached for reference. 
<img width="1072" alt="h5phub-details" src="https://github.com/user-attachments/assets/9c350be5-ba16-4946-bd8e-35427ed0dd22" />
<img width="1072" alt="h5phub-interface" src="https://github.com/user-attachments/assets/d315982a-c011-40e1-af65-a4536bb2b0f6" />

Your expertise would be incredibly helpful! Let me know if you need further details.

**Reply by tmkarlsen** [reply]
Hi, @samumist 

The main reason this is not translatable is that the H5P-hub does not come with language files that are out of the box.
Furthermore, the way these strings are handled proves problematic to maintain translation-wise. This is because these data are stored in the `h5p_libraries_hub_cache` table and updated whenever an update check for content types is made, making them somewhat elusive.

**Reply by samumist** [reply]
Hi @tmkarlsen,

Thanks again for explaining the technical limitations with translation in H5P Hub—I appreciate the insight into the challenges with language files and the `h5p_libraries_hub_cache` updates.

Since `Lumi` has tackled this issue, I’m curious: Is Edlib planning to implement a similar solution? If not, could you point me toward any workarounds or steps my team could take to address this on our end?

Looking forward to your thoughts!
<img width="989" alt="截屏2025-05-05 19 10 12" src="https://github.com/user-attachments/assets/39c8c632-be81-4ad9-acf1-11f2737563e5" />

**Reply by tmkarlsen** [reply]
Hi, @samumist.
This could be solved by ensuring the database handling of some files for the H5P Hub, a solution somewhat similar to the one we have added to the admin interface of the LTI tool for H5P authoring, Content Author.

We do not prioritize this at the moment, but we are happy to receive pull requests on such a solution.

**Reply by samumist** [reply]
Hi @tmkarlsen.
Thanks for pointing that out! Could you share a link to the relevant code or documentation for the database handling solution you mentioned?

We’d love to explore contributing a solution if we’re able to develop one.
